linting/code-linting-eslint.md

```diff
@@ -0,0 +1,143 @@
+
+# Code liniting + ESLint Instructions
+
+### Projected Time
+10-15 minutes
+
+- Lesson: 5-7 min
+- Guided Practice: 5 min (Mock Project)
+- Independed Practice: (Implement it with your projects)
+- Check for Understanding: 1 min
+
+### Prerquisites
+
+Here are topics you must be comfortable with to get the maximum benefit out of this topic:
+
+-  [JavaScript](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md)
+- A project to follow along ([Node.js](https://nodejs.org/en/) (^8.10.0, ^10.13.0, or >=11.10.1))
+
+### Motivation
+
+Linting is a process of using a tool to automatically check your code for potential problems. There are several benefits for doing so for your code:
+
+-  **Highlighting potential errors and bad patterns.** Linters can perform advanced checks to uncover possible errors such as duplicate variables, unreachable code or invalid regular expressions. A warning from the linter will allow you to fix the errors before they even try to run the code.
+-  **Keeping your code style consistent.** It allow you to check you code style for issues like spacing, indentation and placement of braces. Once your team agrees on the coding style, linter can be configured accordingly and checked automatically.
+-  **Enforcing quality.** Well run projects have clear consistent coding conventions, with automated enforcement.If a linting tool is wired into your build process, you may simply prevent the project from starting or being committed into your repository if there are unfixed errors.
+-  **Time saving** At the end it'll only save your time from searching silly bugs and waiting for your Pull Request to be merged due to lack of inconsistency in code style.
+
+### Objectives
+
+**Participants will be able to:**
+
+- Use ESLint seemlessly in their project
+- Integrate ESLint with vscode
+- Use popular style guide suggested by Airbnb
+  
+
+### Specific Things To Learn
+
+- Installation of ESLint
+- Configuration process
+
+### Materials
+
+-  [Improve Your Code With ESLint + VsCode + Airbnb Styleguide](https://www.youtube.com/watch?v=mfGkKlMDfwQ&t=253s) - Colt Steele
+-  [ESLint Quickstart - find errors automatically](https://www.youtube.com/watch?v=qhuFviJn-es) - freeCodeCamp.org
+-  [ESLint + Prettier + VS Code — The Perfect Setup](https://www.youtube.com/watch?v=lHAeK8t94as) - Web Bos
+
+
+### Lesson
+
+ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. In many ways, it is similar to JSLint and JSHint with a few exceptions:
+
+- ESLint uses Espree for JavaScript parsing. 
+- ESLint uses an AST to evaluate patterns in code.
```

Comment:
This is irrelevant to a beginner. Please cut this or move to supplemental materials if there is interesting reading about it.